### PR TITLE
User Feedback Modal

### LINKED ISSUE #217: User feedback modal
## Description

Build the feedback modal for the the user

## Learning objectives

- React TS
- Semantic UI React
- Formik
- SCSS

### PR DESCRIPTION
**Pull request by zorazrr**
## Summary

Create user feedback modal. Admin can leave feedback for user, associated with a specific pitch.
Closes #217

## Changes

- Create user feedback modal

## Screenshots

Screen Shot 2021-12-03 at 8 03 55 PM

### CODE REVIEW

File: client/src/components/Modals/UserFeedback/index.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        reasoning: feedback,
+      };
+      const res = await createUserFeedback(newFeedback);
+      if (!isError(res)) {
+        toast.success('Successfully left feedback', {
+          position: 'bottom-right',
+        });
+        setIsOpen(false);
+      }
+    } else if (isEmpty()) {
```
Comment (amit-sawhney) [inline client/src/components/Modals/UserFeedback/index.tsx:84]: You are calling this function twice. why not just have this in an else block?

File: client/src/components/Modals/UserFeedback/index.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+import { getAggregatedPitch, createUserFeedback, isError } from '../../../api';
+import FieldTag from '../../FieldTag';
+import UserChip from '../../UserChip';
+
+import './styles.scss';
+
+interface UserFeedbackModal extends ModalProps {
+  contributor: IUser;
+  pitchId: string;
+  trigger: ReactElement;
```
Comment (amit-sawhney) [inline client/src/components/Modals/UserFeedback/index.tsx:24]: ModalProps contains this trigger already I believe, no reason to redefine it.

**Review by amit-sawhney — CHANGES_REQUESTED** [review]

File: client/src/components/Modals/UserFeedback/index.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+  const { user } = useAuth();
+  const { getTeamFromId } = useTeams();
+
+  useEffect(() => {
+    if (!isOpen) {
+      return;
+    }
+
+    getContributorTeams();
```
Comment (amit-sawhney) [inline client/src/components/Modals/UserFeedback/index.tsx:44]: Get rid of this lint error :)

**Review by amit-sawhney — APPROVED** [review]

**Review by ayanmallik — APPROVED** [review]
Really cool feature Zora! Great job on utilizing the rating component.

**Review by AndrewLester — APPROVED** [review]

**Pull request merged**